packages/block-library/src/image/edit.native.js

```diff
@@ -45,6 +45,19 @@ import SvgIconRetry from './icon-retry';
 const LINK_DESTINATION_CUSTOM = 'custom';
 const LINK_DESTINATION_NONE = 'none';
 
+const IMAGE_SIZE_THUMBNAIL = 'thumbnail';
+const IMAGE_SIZE_MEDIUM = 'medium';
+const IMAGE_SIZE_LARGE = 'large';
+const IMAGE_SIZE_FULL_SIZE = 'full';
+const DEFAULT_SIZE_SLUG = IMAGE_SIZE_LARGE;
+const sizeOptionLabels = {
+	[ IMAGE_SIZE_THUMBNAIL ]: __( 'Thumbnail' ),
+	[ IMAGE_SIZE_MEDIUM ]: __( 'Medium' ),
+	[ IMAGE_SIZE_LARGE ]: __( 'Large' ),
+	[ IMAGE_SIZE_FULL_SIZE ]: __( 'Full Size' ),
+};
+const sizeOptions = map( sizeOptionLabels, ( label, option ) => ( { value: option, label } ) );
```

Comment:
Nitpicking here, but I'm thinking we should probably have done the opposite:
```
const sizeOptions = [
	{ value: IMAGE_SIZE_THUMBNAIL, label: __( 'Thumbnail' ) },
	{ value: IMAGE_SIZE_MEDIUM, label: __( 'Medium' ) },
	{ value: IMAGE_SIZE_LARGE, label: __( 'Large' ) },
	{ value: IMAGE_SIZE_FULL_SIZE, label: __( 'Full Size' ) },
];
const sizeOptionLabels = sizeOptions.reduce( ( result, { value, label } ) => { result[value] = label; return result; }, {} );
```
Or rather, since this is only called whenever the value of the picker changes, I don't think precomputing `sizeOptionLabels` is even worth it:
``` 
<BottomSheet.PickerCell
    value={ sizeOptions.find( { value: sizeSlug || DEFAULT_SIZE_SLUG } ) }
    ...
/>
```

Reply:
I appreciate the nitpicking. I need to work on considering lodash and Array functions in these cases. Still transitioning from Java/Kotlin world :-P.  Let me know if you want me to update this.

Reply:
Aren't all those constants shared between web and mobile? I would expect they exist already and could be reused.

Reply:
In this case, `imageSize` comes from block-editor settings: 
https://github.com/WordPress/gutenberg/blob/de0cc66a4f2545ead0ddef544d434349f371078c/packages/block-library/src/image/edit.js#L938

https://github.com/WordPress/gutenberg/blob/de0cc66a4f2545ead0ddef544d434349f371078c/packages/block-library/src/image/edit.js#L554-L555

I did a small test and we have this available on mobile too. So we can reuse it 👍